src/main/java/io/stargate/sgv2/jsonapi/service/schema/SimilarityFunction.java

```diff
@@ -1,27 +1,44 @@
 package io.stargate.sgv2.jsonapi.service.schema;
 
 import io.stargate.sgv2.jsonapi.exception.ErrorCodeV1;
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * The similarity function used for the vector index. This is only applicable if the vector index is
  * enabled.
  */
 public enum SimilarityFunction {
-  COSINE,
-  EUCLIDEAN,
-  DOT_PRODUCT,
-  UNDEFINED;
+  COSINE("cosine"),
+  EUCLIDEAN("euclidean"),
+  DOT_PRODUCT("dot_product"),
+  UNDEFINED("undefined");
+
+  private String metric;
+  private static Map<String, SimilarityFunction> functionMap = new HashMap<>();
```

Comment:
naming: `static` fields should typically be ALL_CAPS (as per Java naming convention).